src/shared/components/post/post-form.tsx

```diff
@@ -954,3 +960,21 @@ export class PostForm extends Component<PostFormProps, PostFormState> {
     this.setState({ bypassNavWarning: false });
   }
 }
+
+export function filterCommunitySelection(
+  comms: CommunityView[],
+  my_user?: MyUserInfo,
+): CommunityView[] {
+  const follows = my_user?.follows.map(c => c.community.id);
+  return (
+    comms
+      // filter out comms where only mods can post, unless current user is mod
+      .filter(c => !c.community.posting_restricted_to_mods || c.can_mod)
+      // filter out private comms unless the current user follows it
+      .filter(
+        c =>
+          c.community.visibility !== "Private" ||
+          follows?.includes(c.community.id),
```

Comment:
Doesn't the follow state have to be approved? IE you can be trying to follow a private community, and `Pending` will come back in your follow list.

Reply:
I added a backend filter for that: https://github.com/LemmyNet/lemmy/pull/6003/files#diff-4f4537d4bbffbcd3b0a6f0bb920ffb8439ba34f77017b89ebe6291190413ac2aR82